core/string/uminus_spec.rb

```diff
@@ -17,5 +17,18 @@
       output.frozen?.should == true
       output.should == 'foo'
     end
+
+    ruby_version_is "2.5" do
+      it "returns the same object for equal unfrozen strings" do
+        origin = -"this string is frozen"
+        dynamic = -%w(this string is frozen).join(' ')
+
```

Comment:
It would be interesting to not call `-@` on the dynamic part and first check `origin.should_not equal dynamic`.
Then `(-dynamic).should equal origin`.